src/Microsoft.ML.AutoML/ColumnInference/ColumnInferenceApi.cs

```diff
@@ -56,7 +56,8 @@ internal static class ColumnInferenceApi
                 AllowSparse = splitInference.AllowSparse,
                 AllowQuoting = splitInference.AllowQuote,
                 HasHeader = hasHeader,
-                TrimWhitespace = trimWhitespace
+                TrimWhitespace = trimWhitespace,
+                ReadMultilines = true // MYTODO: is it ok to hardcode this? it is necessary for my test to pass
```

Comment:
So I am not familiar at all with AutoML code, but in order to add a test reproducing the scenario in #4460 I had to add this in here for it to work. Problem is that these methods in AutoML and _several_ more methods there would require to either hardcode this or add more parameters to several methods to have a readMultilines parameter.

This is a problem because ReadMultilines is set as false by default, and we would need to manually set it as true anywhere where we'd want AutoML to have this behavior. So I am not sure what would be the best way to handle this from AutoML side.

Reply:
I expect it's better to sweep over the `ReadMultilines` flag. The same as we do with the other flags for `TextLoader` ([src](https://github.com/dotnet/machinelearning/blob/e50c4d20012e0d62852f404ae443afca7dad043e/src/Microsoft.ML.AutoML/ColumnInference/TextFileContents.cs#L51-L52)).

Are there any datasets (or rows) which would be unreadable with `ReadMultilines = true`?

Reply:
As I mention here: [https://github.com/dotnet/machinelearning/pull/5125#discussion_r425034529](https://github.com/dotnet/machinelearning/pull/5125#discussion_r425034529) , if the CSV file contains a quoted field which has a " at the beginning of the field, but it never closes it (which would technically be invalid format), then the file won't be read correctly and some problems may arise from it.

For example, our "wikipedia-detox-250-line-data.tsv" has this error in [line 83](https://github.com/dotnet/machinelearning/blob/3da9e1e7a64f3485b811b99eaab05276a310192c/test/data/wikipedia-detox-250-line-data.tsv#L83). The sentiment text field there opens with " but it never closes that quotation, so in that dataset the Multilinereader doesn't load that correctly.

---
In reply to: [425713757](https://github.com/dotnet/machinelearning/pull/5125#discussion_r425713757) [](ancestors = 425713757)

Reply:
Regarding your suggestion of simply sweeping over the flags in TrySplitColumns, it doesn't look that would affect the InferColumns method I'm mentioning in this thread (it looks to me the InferColumns method isn't called when calling TrySplitColumns). As you can see in the [ColumnInferenceApi\([https://github.com/dotnet/machinelearning/blob/e50c4d20012e0d62852f404ae443afca7dad043e/src/Microsoft.ML.AutoML/ColumnInference/ColumnInferenceApi.cs](https://github.com/dotnet/machinelearning/blob/e50c4d20012e0d62852f404ae443afca7dad043e/src/Microsoft.ML.AutoML/ColumnInference/ColumnInferenceApi.cs)) there are _several_ methods which have multiple parameters related to TextLoader, and it seems I'd need to add a new parameter to all of them. But my question is that I wouldn't know if it's necessary to add it in all of them to make the readMultiline parameter accesible to ModelBuilder. Furthermore, it seems these methods are called in the [AutoCatalog.cs](https://github.com/dotnet/machinelearning/blob/e50c4d20012e0d62852f404ae443afca7dad043e/src/Microsoft.ML.AutoML/API/AutoCatalog.cs#L147-L210) which also exposes some public methods with parameters related to TextLoader. And it would again seem that I'd need to add a new readMultiline parameter there... but I guess won't be able to do that since that is a public API, and adding parameters is considered a breaking API change.

So, in general, I don't know which criteria to use in order to know where to add a readMultiline parameter in order to make it accessible for ModelBuilder. Is there any other place besides ColumnInferenceAPI and TrySplitColumns?

---
In reply to: [425713757](https://github.com/dotnet/machinelearning/pull/5125#discussion_r425713757) [](ancestors = 425713757)

Reply:
As discussed in here: I won't change anything in AutoML.NET code, and will leave this up to ModelBuilder team.
[https://github.com/dotnet/machinelearning/pull/5125#discussion_r426017274](https://github.com/dotnet/machinelearning/pull/5125#discussion_r426017274)

---
In reply to: [425726377](https://github.com/dotnet/machinelearning/pull/5125#discussion_r425726377) [](ancestors = 425726377,425713757)